core-tests/shared/src/test/scala/zio/ChunkSpec.scala

```diff
@@ -1,40 +1,398 @@
 package zio
 
-import zio.test.Assertion.equalTo
-import zio.test.{ assert, suite }
+import zio.random.Random
+import zio.test.Assertion._
+import zio.test.TestAspect.exceptScala211
+import zio.test._
 
 object ChunkSpec extends ZIOBaseSpec {
 
+  case class Value(i: Int) extends AnyVal
+
+  import ZIOTag._
+
+  val intGen = Gen.int(-10, 10)
+
+  def toBoolFn[R <: Random, A]: Gen[R, A => Boolean] =
+    Gen.function(Gen.boolean)
+
+  def tinyChunks[R <: Random, A](a: Gen[R, A]): Gen[R, Chunk[A]] =
+    Gen.chunkOfBounded(0, 3)(a)
+
+  def smallChunks[R <: Random, A](a: Gen[R, A]): Gen[R with Sized, Chunk[A]] =
+    Gen.small(Gen.chunkOfBounded(0, _)(a))
```

Comment:
`Gen.small` generates a random value from a small size distribution. `Gen.chunkOfBounded` generates values between the minimum and the maximum. So by combining them in this way I think you are biasing the size downward a little bit. I you just do `Gen.small(Gen.chunkOfN(_)(a))` you will get the "small" distribution of sizes.